File: Frontend/src/API/Contracts/EmployeeResponse.ts
import { Salary } from "./CreateEmployeeRequest";

export interface EmployeeResponse {
    id: number;
    name: string;
    jobTitle: string;
    dateOfJoining: string;
    departmentName: string;
    lineManagerId: number;
    address: string;
    salary: Salary;
}

export interface CommentResponse {
    commentId: number;
    recipientEmployeeId: number;
    text: string;
    dateCreated: string;
    authorName: string;
}

File: Backend/Application/Contracts/Employee/EmployeeResponse.cs
using Domain;

namespace Application.Contracts.Employee;

public class EmployeeResponse
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string JobTitle { get; set; }
    public DateTime DateOfJoining { get; set; }
    public string DepartmentName { get; set; }
    public int LineManagerId { get; set; }
    public string Address { get; set; }
    public Salary Salary { get; set; } = new();
}
